### DISCUSSION TITLE
Do I need to worry about these? When upgrading from v2.2.0 to v2.3.0

### DISCUSSION THREAD

**Discussion by itwye** [goharbor/harbor#17075] [Q&A]
hi guys

I'm about to upgrade from harbor 2.2.0 to harbor 2.3.0, while watching the [upgrade documentation for 2.3.0](https://goharbor.io/docs/2.3.0/administration/upgrade/), I noticed two things

1. > "In v2.0 the metadata of artifacts are stored in database, when Harbor is started for the first time after the upgrade it will walkthrough the artifacts in registry storage to extract metadata of the artifacts into database. This process may take relatively long time if there are large number of artifacts in the registry, especially when registry is configured to use external storage like S3. During this process the Harbor API will not be responsive please check the log of harbor-core and registry to monitor the process."

I am currently using local disk storage, and the disk space used by the image is about 6T. According to the above statement, after upgrading to v2.3.0, will it take a lot of time to "extract metadata of the artifacts into database" at the first startup? , how long will the process take?

2. > "Harbor v2.3 uses PostgreSQL v13.2. During an upgrade, Harbor will remove the old PostgreSQL data and migrate it to new destination to compatible new version PostgreSQL. You must backup your data before upgrading to v2.3.0."

I am currently using an external database (psql v9.6.15), according to the above, do I need to upgrade the psql database version to v13.2 before upgrading?

**Comment by zyyw** [comment]
It may indeed take some time to complete the migration, presuming there are lots of artifacts in your harbor given the disk space taken by the. imaged is about 6T.

It does need to upgrade the pgsql.

Please DO back-up your data first before starting to migrate.
